extraction_tools_wiki/wiki_page_titles.py

```diff
@@ -31,20 +31,19 @@
 class WikiPageTitles:
     """This class handles extraction of wiki page titles by category using an OSRS Wiki API query.
 
-    :param :obj:`list` of :obj:`str` categories: A list of OSRS Wiki categories.
-    :param str out_file_name: The file name used for exporting the wiki page titles to JSON.
-    :param str user_agent: A custom user-agent name to be used for the API request.
-    :param str user_email: A custom user-agent email to be used for the API request.
+    :param base_url: The OSRS Wiki URL used for API queries.
+    :param categories: A list of OSRS Wiki categories.
+    :param user_agent: A custom user-agent name to be used for the API request.
+    :param user_email: A custom user-agent email to be used for the API request.
     """
-    def __init__(self, categories, out_file_name, user_agent, user_email):
-        self.base_url = "https://oldschool.runescape.wiki/api.php"
-        self.page_titles = dict()
+    def __init__(self, base_url: str, categories: list, user_agent: str, user_email: str):
+        self.base_url = base_url
         self.categories = categories
-        self.out_file_name = out_file_name
         self.custom_agent = {
             'User-Agent': user_agent,
             'From': user_email
         }
+        self.page_titles = dict()  # type: dict

```

Comment:
You will want to use the typing module for list, dict, set, etc:

https://docs.python.org/3/library/typing.html#typing.Dict

```python
self.page_titles: Dict = dict()
```

The `Dict` can also take the type of its keys and values, for instance if we have string keys to int values:
```python
self.page_titles: Dict[str, int] = dict()
```

The more information we have, the better the typing validation we can do with `mypy`.